Свалова Настя

SvalovaNastya opened the pull request:
[Посмотреть решение](https://rawgit.com/SvalovaNastya/markup-task-4/master/index.html)

Commit:
сделана статья в газете

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

Victoria-Vladimirova commented:
**Chrome**

При разрешении 1920х1080 часть картинки переходит в другой столбик

![2016-10-31 20 08 10](https://cloud.githubusercontent.com/assets/14944973/19863747/40545c7a-9fb7-11e6-9bfe-1f4cf6274399.png)

**FF**

Разваливается в FF

![2016-10-31 20 14 35](https://cloud.githubusercontent.com/assets/14944973/19863880/c5c8e95c-9fb7-11e6-9b5c-01eab624bc50.png)

**Edge**

Исчез последний столбик

![2016-10-31 22 00 50](https://cloud.githubusercontent.com/assets/14944973/19863952/0775f41c-9fb8-11e6-9d43-e05b6258db8a.png)

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,106 @@
+    <header>The <img src="images/The_Times_logo_white_bg.png" alt="times logo"> Times</header>
+
+    <main>
+        <h3>Royals honour Queen Mother in Charles’s model village</h3>
```
Если здесь заголовок третьего уровня, то где тогда заголовки уровней 1 и 2?

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,106 @@
+        <h3>Royals honour Queen Mother in Charles’s model village</h3>
+
+        <div class="img_block">
+            <img id="monument_img" src="images/monument.jpg" alt="monument">
```
Нужно добавить всем картинкам title. Значение alt должно отвечать на вопрос «Что изображено на картинке?»

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,106 @@
+
+        <div class="img_block">
+            <img id="monument_img" src="images/monument.jpg" alt="monument">
+            <div class="caption">Prince Charles said: <q>It is with immense pride and in
```
Наверно, здесь тоже надо переносить тест, как ты это делаешь для `<p></p>`

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,106 @@
+            to make the scheme a&nbsp;real&shy;ity.</q>
+        </p>
+
+        <div class="img_inline">
```
Неудачное название для класса. 

На лекции по html рассказывали про figure и figcaption, они сюда хорошо подходят.

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,158 @@
+@font-face
+{
+    font-family: 'Abhaya Libre';
+    src: url('fonts/Abhaya_Libre/AbhayaLibre-ExtraBold.ttf');
```
Нужно подключить woff

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,158 @@
+
+header
+{
+    font-family: 'Abhaya Libre';
```
font-family и font-size можно объединить

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,158 @@
+{
+    display: block;
+    width: 50%;
+    float: left;
```
float не используем

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,158 @@
+    column-span: all;
+    display: block;
+    float: none;
+    /*width: 80rem;*/
```
тут что-то лишнее

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,158 @@
+
+h3
+{
+    font-family: 'Indie Flower';
```
Обычно ещё указывают шрифты или семейство шрифтов, которые будут подключены, если твой Indie Flower не загрузится

SvalovaNastya replied:
как он может не загрузиться, если он тут рядышком лежит?

Victoria-Vladimirova replied:
Файл со шрифтом большой, он может не загрузиться из-за плохого канала связи или загрузиться с задержкой и применится позже.

Об этом рассказывал Олег на 3й лекции "Шрифты и текст". Полезно также смотреть примеры в лекции.

Victoria-Vladimirova commented:
Задание выполнено не полностью. Нет текста, который располагается вертикально 

> - Изредка текст пишут:
>   В
>   О
>   Т
>   
>   Т
>   А
>   К

Victoria-Vladimirova commented:
:tomato:

Commit:
задание со звездой + вертикальный текст

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Commit:
замечания по ревью + поправила линт.

Не сделано: в firefox до сих пор не по колонкам, в chome до сих пор иногда картинка переносится

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

Commit:
замечани по ревью: добавлен маленький шрифт, и убрала лишние строчки в css

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

Commit:
добавлены title

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

Commit:
добавила дефолтные шрифты

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

Commit:
шрифты

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Commit:
я ваще не понимаю, что ему от меня надо

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Commit:
упоротый линт

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,129 @@
+        <main>
+        <h1>Royals honour Queen Mother in Charles’s model village</h1>
+
+        <div class="img_block">
```
`figure`

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,129 @@
 <body>
+    <header>The <img src="images/The_Times_logo_white_bg.png" alt="times logo"> Times</header>
+
+        <input type="radio" class="checkbox" name="font_size" value="large">
```
Здесь не будет отступа. `header` и  `input` на одном уровне

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,129 @@
+        <span class="checkbox_caption">Сделать черный на белом</span><br>
+
+        <main>
+        <h1>Royals honour Queen Mother in Charles’s model village</h1>
```
а вот здесь не хватает отступа

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,129 @@
+            <em>bat&shy;tling against the tide</em>.
+        </p>
+
+        <p>
```
`blockquote`

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,129 @@
+        </p>
+
+        <p>
+            <q>Bat&shy;tling against the tide is an un&shy;com&shy;fort&shy;able
```
Очень длинная цитата. Больше подойдет blockquote

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,129 @@
+        </p>
+
+        <p class="end">
+            T h e  E n d!
```
Не нужно переносить, т.к строка меньше 110 символов. 
Также здесь не нужны пробелы лишние.

SvalovaNastya replied:
пробелы нужны. без них не всегда переносится

Victoria-Vladimirova replied:
Возможно ты что-то делаешь не так :)

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,129 @@
+            stands as a beacon to the way for&shy;ward.</q>
+        </p>
+
+        <p class="end">
```
Плохое название для класса. Непонятно из названия, к чему он относится.

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,129 @@
+            to make the scheme a&nbsp;real&shy;ity.</q>
+        </p>
+
+        <figure class="img_inline">
```
Было замечание исправить название. 

Почему inline? Ты в css присваиваешь `display: block` Название класса должно состоять только из `a-z` и `-`

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,129 @@
+        <h1>Royals honour Queen Mother in Charles’s model village</h1>
+
+        <div class="img_block">
+            <img id="monument_img" src="images/monument.jpg"
```
зачем в данном случае картинке присвоен id?

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,209 @@
+}
+
+figure img
+{
```
Картинкам нужно указать ширину и высоту. 
Так браузер будет знать размер картинки и зарезервирует для неё место при загрузке страницы. Иначе компоновка страницы будет меняться во время загрузки.

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,209 @@
+{
+    padding: 1%;
+    margin: 0;
+    height: 500%;
```
почему такая высота указана?

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,209 @@
+{
+    column-span: all;
+    width: 100%;
+    border: 3px groove darkslategray;
```
Цвета лучше указывать используя hex-код

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,209 @@
+    -moz-column-count: 4;
+    column-gap: 5px;
+    -moz-column-gap: 5px;
+    column-rule: 3px groove darkslategray;
```
Безвендорное правило должно идти после правил с вендорными префиксами

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,209 @@
+    src: url('fonts/Lobster/Lobster-Regular.ttf'), url('fonts/Lobster/Lobster-Regular.woff');
+}
+
+input[name='font_size'][value='large']:checked ~ main
```
Почему бы вместо `input[name='font_size']` не сделать класс?

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,209 @@
+@font-face
+{
+    font-family: 'Fredoka One';
+    src: url('fonts/Fredoka_One/FredokaOne-Regular.ttf'), url('fonts/Fredoka_One/FredokaOne-Regular.woff');
```
```
src: url('fonts/Fredoka_One/FredokaOne-Regular.ttf'),
     url('fonts/Fredoka_One/FredokaOne-Regular.woff');
```

SvalovaNastya replied:
так lint ругается. невозможно нормально сделать

mokhov replied:
А чего пишет?

SvalovaNastya replied:
`4:61  ×  Expected single space after ","    value-list-comma-space-after`

независимо от того, ставлю ли я пробелы

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,209 @@
+
+header
+{
+    font-family: 'Abhaya Libre', 'Arial';
```
Arial - шрифт без засечек, а Abhaya Libre - с засечками. Лучше указывать соответствующую замену в виде шрифта (times new roman) и/или семейства (serif)

`font-family: 'Abhaya Libre', serif`

В остальных местах тоже надо поправить

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,209 @@
+{
+    font-family: 'Abhaya Libre', 'Arial';
+    text-transform: uppercase;
+    font-size: 5rem;
```
`font-family` и `font-size` можно объединить и написать в одну строчку как `font`

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,209 @@
+{
+    font-family: 'Playball', 'Arial';
+    text-indent: 4%;
+    font-size: 120%;
```
Почему в процентах?

SvalovaNastya replied:
проценты же тоже нормальная практика. нам не говорили, что это плохо

Victoria-Vladimirova replied:
Нет, мне интересно, почему почти везде em, а тут в одном месте проценты.

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,209 @@
+figcaption
+{
+    font-family: 'Rock Salt', 'Arial';
+    font-size: .6rem;
```
где-то в em указан размер, а где-то в rem. Почему так?

SvalovaNastya replied:
потому что где-то я хочу указывать от родительского, а где-то просто в каких-то, зависящих от экрана (не константные пискели) величинах

Victoria-Vladimirova commented:
🍅

Commit:
замечани по ревью

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

SvalovaNastya commented:
🍏

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,204 @@
+    font-family: 'Trirong';
+}
+
+input[id='color']:checked ~ main
```
`#color:checked ~ main`

SvalovaNastya replied:
id нельзя использовать в css

Victoria-Vladimirova replied:
но ты же используешь)

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,127 @@
+    <header>The <img src="images/The_Times_logo_white_bg.png" alt="times logo"> Times</header>
+
+    <input type="radio" class="checkbox" name="font_size" value="large">
+    <span class="radio_caption">Сделать шрифт больше</span><br>
```
`label`

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,127 @@
 <body>
+    <header>The <img src="images/The_Times_logo_white_bg.png" alt="times logo"> Times</header>
+
+    <input type="radio" class="checkbox" name="font_size" value="large">
```
checkbox это что-то совсем не то

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,204 @@
+    font-size: 1.3em;
+}
+
+input[name='font_size'][value='small']:checked ~ main
```
тут уже можно было свой класс использовать

SvalovaNastya replied:
у меня не работает так

Victoria-Vladimirova replied:
Напиши мне свой селектор

SvalovaNastya replied:
заработало :(

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,127 @@
+    <input type="radio" class="checkbox" name="font_size" value="small">
+    <span class="radio_caption">Сделать шрифт меньше</span><br>
+    <input type="radio" class="checkbox" name="font_size" value="normal">
+    <span class="radio_caption">Сделать шрифт как был</span><br>
```
ты уже знаешь о `display` (раз пишешь у себя в css), тогда зачем тебе здесь `<br>`?

SvalovaNastya replied:
я не очень хорошо знала display, когда писала.

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,127 @@
+            res&shy;id&shy;en&shy;tial areas, shops and leis&shy;ure fa&shy;cil&shy;it&shy;ies
+            to cre&shy;ate a walk&shy;able com&shy;munity.
+        </p>
+
```
не оставляй лишних пустых строк

SvalovaNastya replied:
мне так визуально приятнее смотреть на код

Victoria-Vladimirova replied:
Почему тогда не все параграфы разделены пустой строкой?
Проверять так не очень удобно

SvalovaNastya replied:
были все, кроме одного

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,127 @@
+    <title>The Times</title>
 </head>
 <body>
+    <header>The <img src="images/The_Times_logo_white_bg.png" alt="times logo"> Times</header>
```
title?

Victoria-Vladimirova reviewed (commented):
Часть замечаний не была исправлена. Появились еще замечания

Victoria-Vladimirova commented:
🍅

Commit:
замечания

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Commit:
lint + замечания

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Commit:
Какая-то дичь с тэгами

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

SvalovaNastya commented:
🍏

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,208 @@
+{
+    font-family: 'Cookie Regular', fantasy;
+    font-size: 4em;
+    color: crimson;
```
hex

mokhov replied:
Так и не исправила :(

Victoria-Vladimirova commented on index.css:
```diff
@@ -0,0 +1,208 @@
+    font-size: 1.3em;
+}
+
+input[value='small'].radio:checked ~ main
```
`.radio[value='small']:checked ~ main`

Victoria-Vladimirova commented on index.html:
```diff
@@ -1,8 +1,108 @@
+            had been out of fa&shy;vour, de&shy;scrib&shy;ing how he had been
+            <em>bat&shy;tling against the tide</em>.
+        </p>
+        <p>
```
Тег не закрыт, потому что тесты валятся?  `blockquote` нельзя писать внутри `p` 
Браузер сделает вот так: 

```
<p> </p>
<blockquote>Твой текст</blockquote>
```

Victoria-Vladimirova commented:
:rocket:

mokhov commented on index.css:
```diff
@@ -0,0 +1,208 @@
+    color: #900;
+    word-spacing: 1rem;
+    font-size: 2rem;
+    line-height: 1em;
```
Вообще не понятно почему здесь em?

mokhov commented on index.css:
```diff
@@ -0,0 +1,208 @@
+
+p
+{
+    text-indent: 2%;
```
Почему в %?

SvalovaNastya replied:
я все же не понимаю, когда нужно какие единицы измерения писать. Я запуталась

mokhov replied:
Для шрифтов чаще всего используются px, em и rem. Соответственно для всех текстовых свойств это же правило действует

mokhov commented on index.css:
```diff
@@ -0,0 +1,208 @@
+    word-wrap: break-word;
+    margin-left: 35%;
+    color: inherit;
+    font: 2em inherit;
```
Нет смысла указывать inherit, оно и так наследуется

mokhov commented on index.css:
```diff
@@ -0,0 +1,208 @@
+    text-align: center;
+    word-wrap: break-word;
+    margin-left: 35%;
+    color: inherit;
```
Тут то же самое

mokhov commented on index.html:
```diff
@@ -1,8 +1,108 @@
 </head>
 <body>
+    <header>The <img src="images/The_Times_logo_white_bg.png" alt="times logo" title="times logo"> Times</header>
+    <input type="radio" class="radio" name="font_size" value="large" id="larger">
```
@Victoria-Vladimirova и ты не попросила вынести всё это на красивую плашку?

Victoria-Vladimirova replied:
@mokhov Если мне не хочется орать от увиденного, то я не навязываю свой вкус студентам

mokhov replied:
@Victoria-Vladimirova дома поговорим!

mokhov commented on index.html:
```diff
@@ -1,8 +1,108 @@
+            plan&shy;ning and de&shy;vel&shy;op&shy;ment in this coun&shy;try. At a time
+            when there is a clam&shy;our for new hous&shy;ing I be&shy;lieve that Pound&shy;bury
+            stands as a beacon to the way for&shy;ward.</blockquote>
+        <p class="vertical-block">T h e  E n d!</p>
```
Забавный способ :)

mokhov commented on index.html:
```diff
@@ -1,8 +1,108 @@
+            hous&shy;ing. The es&shy;tate also provides jobs for 2,100 people and is
+            home to 185 busi&shy;nesses.
+        </p>
+        <p>
```
Этот p и не нужен и не закрыт :(

SvalovaNastya replied:
если убрать совсем, то первая буква некрасивой будет

mokhov reviewed (changes requested):
🍅

Commit:
убрала inherit и оставила параграф, но закрыла его

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

SvalovaNastya commented:
🍏

mokhov commented on index.css:
```diff
@@ -0,0 +1,207 @@
+{
+    column-span: all;
+    width: 100%;
+    border: 3px groove #2f4f4f;
```
И здесь hex

SvalovaNastya replied:
а это разве не hex?

mokhov commented on index.css:
```diff
@@ -0,0 +1,207 @@
+    border: 3px groove #2f4f4f;
+    padding: 20px 0;
+    margin: 0;
+    align-content: center;
```
А где ты это свойство нагуглила, кстати? Оно ведь не работает

SvalovaNastya replied:
это было так давно) я не помню. Действительно не работает

mokhov commented on index.css:
```diff
@@ -0,0 +1,207 @@
+    margin: 0 20px;
+}
+
+figure.img-full-line figcaption.caption-full-line
```
Если есть классы, то писать теги почти всегда избыточно

mokhov commented on index.css:
```diff
@@ -0,0 +1,207 @@
+{
+    padding: 1%;
+    margin: 0;
+    width: 98%;
```
В CSS есть такое правило, что ширину лучше менять с дефолтной только в ситуациях, когда иначе нельзя поступить. Например здесь можно задать другое CSS свойство и не трогать ширину. Про это рассказывали на лекции

SvalovaNastya replied:
хм, на самом деле тут ничего не надо задавать

mokhov commented on index.css:
```diff
@@ -0,0 +1,207 @@
+figure.img-column img
+{
+    width: 100%;
+    display: block;
```
Зачем нужен display: block в данном случае?

SvalovaNastya replied:
без него в Edge плывет последний абзац

mokhov commented on index.html:
```diff
@@ -1,8 +1,108 @@
+        </p>
+        <p>
+            An&shy;drew Hamilton, de&shy;vel&shy;op&shy;ment dir&shy;ect&shy;or for the
+            Duchy of Corn&shy;wall, told Prince Charles:</p> <blockquote>You, sir, single-handedly
```
Тогда вот здесь отбей добавь перенос чтобы было красиво

mokhov reviewed (changes requested):
🍅

Commit:
еще немного замечаний

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

SvalovaNastya commented:
🍏 😩
